app2/src/routes/faucet/Faucet.svelte

```diff
@@ -4,38 +4,171 @@ import Card from "$lib/components/ui/Card.svelte"
+import { Turnstile } from "svelte-turnstile"
+import request from "graphql-request"
+import { writable } from "svelte/store"
+import { faucetUnoMutationDocument } from "$lib/queries/faucet"
+import { URLS } from "$lib/constants"
+import { Data } from "effect"
+import { bech32, bytes } from "@scure/base"
+
+// Define the faucet state type using Data.TaggedEnum.
+type FaucetProcessState = Data.TaggedEnum<{
+  Idle: {}
+  Verifying: {}
+  Verified: { token: string }
+  Submitting: { token: string }
+  Success: { message: string }
+  Failure: { error: string }
+}>
+
+// Create the tagged enum instance.
+const FaucetProcess = Data.taggedEnum<FaucetProcessState>()
+
+// Initialize the faucet process state to Idle.
+const faucetProcess = writable(FaucetProcess.Idle())
+
+// Variables for managing the Turnstile component.
+let resetTurnstile: () => void
+let showTurnstile = false
+
+// When the user clicks "Claim", trigger verification.
+// For now, we bypass verification and use a dummy token.
+const startVerification = () => {
+  faucetProcess.set(FaucetProcess.Verifying())
+  showTurnstile = true
+  resetTurnstile?.() // resets/retriggers the Turnstile if available
+}
+
+// Callback for successful Turnstile captcha.
+const handleTurnstileCallback = (
+  e: CustomEvent<{ token: string; preClearanceObtained: boolean }>
+) => {
+  const token = e.detail.token
+  faucetProcess.set(FaucetProcess.Verified({ token }))
+  // Immediately submit the faucet request.
+  submitFaucetRequest(token)
+}
+
+// Callback for a Turnstile error.
+const handleTurnstileError = (e: CustomEvent<{ code: string }>) => {
+  faucetProcess.set(FaucetProcess.Failure({ error: `Verification error: ${e.detail.code}` }))
+  showTurnstile = false
+}
+
+// Submit the faucet request and update the state accordingly.
+const submitFaucetRequest = async (token: string) => {
+  faucetProcess.set(FaucetProcess.Submitting({ token }))
+  try {
+    let wallet_addr = ""
+    if (Option.isSome(wallets.cosmosAddress)) {
+      const words = bech32.toWords(bytes("hex", wallets.cosmosAddress.value.slice(2)))
+      wallet_addr = bech32.encode("union", words)
+    }
+    const result = await request(URLS().GRAPHQL, faucetUnoMutationDocument, {
+      chainId: "union-testnet-10",
+      denom: "muno",
+      address: wallet_addr,
+      captchaToken: token
+    })
+    console.info("faucet result is:", result)
+    if (!result || !result.drip_drop || !result.drip_drop.send) {
+      faucetProcess.set(FaucetProcess.Failure({ error: "Empty faucet response" }))
+      showTurnstile = false
+      return
+    }
+
+    if (result.drip_drop.send.startsWith("ERROR")) {
+      faucetProcess.set(
+        FaucetProcess.Failure({ error: `Error from faucet: ${result.drip_drop.send}` })
+      )
+      showTurnstile = false
+      return
+    }
+
+    faucetProcess.set(FaucetProcess.Success({ message: result.drip_drop.send }))
+    showTurnstile = false
+  } catch (error) {
+    console.info("error is:", error)
```

Comment:
Never use trycatch, it is fully banned across all union products. instead use effects to handle errors and use effect to make httpcalls that can fail